src/impl.ml

```diff
@@ -715,8 +715,18 @@ let write_stream common s destination _source_protocol destination_protocol prez
       (* TODO: https is not currently implemented *)
       let port = match Uri.port uri' with None -> (if use_ssl then 443 else 80) | Some port -> port in
       let host = match Uri.host uri' with None -> failwith "Please supply a host in the URI" | Some host -> host in
-      Lwt_unix.gethostbyname host >>= fun host_entry ->
-      let sockaddr = Lwt_unix.ADDR_INET(host_entry.Lwt_unix.h_addr_list.(0), port) in
+      let host =
+        match String.split_on_char '[' host with
+        | ""::[elem] ->
```

Comment:
Instead of using the two matches, [Scanf](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Scanf.html) can be used instead:

```ocaml
let host = Scanf.ksscanf host (fun _ _ -> host) "[%s@]" (fun elem -> elem) in
```

ksscanf allows to use a function in case the match fails, and the magic `@` is needed to make it less eager to swallow characters, it's explained in the link.

Reply:
Done :+1: